addons/loyalty/models/loyalty_program.py

```diff
@@ -474,10 +474,12 @@ def create_from_template(self, template_id):
         program = self.create(template_values[template_id])
         action = {}
         if self.env.context.get('menu_type') == 'gift_ewallet':
-            action = self.env['ir.actions.act_window']._for_xml_id('loyalty.loyalty_program_discount_loyalty_action')
-        else:
             action = self.env['ir.actions.act_window']._for_xml_id('loyalty.loyalty_program_gift_ewallet_action')
-        action['views'] = [[False, 'form']]
+            action['views'] = [[False, 'form']]
+        else:
+            action = self.env['ir.actions.act_window']._for_xml_id('loyalty.loyalty_program_discount_loyalty_action')
+            view_id = self.env.ref('loyalty.loyalty_program_view_form').id
+            action['views'] = [[view_id, 'form']]
```

Comment:
I understand this part of the commit, but I don't get why the remaining diff is in this commit :thinking:

Reply:
Commit message has been updated